tools/src/main/java/io/narayana/db/CIPostgreAllocator.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2016, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package io.narayana.db;
+
+import org.apache.commons.lang.StringUtils;
+
+/**
+ * This is a remote database the TS does not have any control over.
+ * The database is always present, always allocated.
+ *
+ * @author <a href="mailto:karm@redhat.com">Michal Karm Babacek</a>
+ */
+public class CIPostgreAllocator extends Allocator {
+
+    CIPostgreAllocator() {
+        // Use getInstance
+    }
+
+    public DB allocateDB(final int expiryMinutes) {
+        final String versionPostgreSQLDriver = System.getProperty("version.postgresql");
+        if (StringUtils.isEmpty(versionPostgreSQLDriver)) {
+            throw new IllegalArgumentException("version.postgresql must not be empty");
+        }
+        final String user = System.getProperty("pgsql.user");
+        if (StringUtils.isEmpty(user)) {
+            throw new IllegalArgumentException("pgsql.user must not be empty");
+        }
+        final String password = System.getProperty("pgsql.password");
+        if (StringUtils.isEmpty(password)) {
+            throw new IllegalArgumentException("pgsql.password must not be empty");
+        }
+        final String servername = System.getProperty("pgsql.servername");
+        if (StringUtils.isEmpty(servername)) {
+            throw new IllegalArgumentException("pgsql.servername must not be empty");
+        }
+        final String portnumber = System.getProperty("pgsql.portnumber");
+        final int port;
+        if (StringUtils.isEmpty(portnumber)) {
+            throw new IllegalArgumentException("pgsql.portnumber must not be empty");
+        } else {
+            port = Integer.parseInt(portnumber);
+            if (port > 65535 || port < 1025) {
```

Comment:
I know it's a bad behaviour to run whatever at port lower to 1025 but you never know. I would personally vote not checking here the range and leave the user to handle it himself.